### DISCUSSION TITLE
upython v 1.23 availability

### DISCUSSION THREAD

**Discussion by krismc** [micropython/micropython#16483] [General]
Does anybody know where I can download upython v1.23 ? The github downloads page seems to have only the latest version?

**Comment by davefes** [comment]
Which bit of hardware?  ESP32 go back to V1.12

**Reply by krismc** [reply]
Looking for a v1.23 for an esp32s2, can find only v1.24 on github

**Comment by mattytrentini** [comment]
Take a look at MicroPython's firmware Download site:

https://micropython.org/download/ESP32_GENERIC_S2/

**Reply by krismc** [reply]
Ah, there they are. Tnx.
